pkg/roundtripper/roundtripper.go

```diff
@@ -0,0 +1,273 @@
+package roundtripper
```

Comment:
question: Would it be worth either moving this code to the common pkg with `http.go` or move that code here? I find it a little confusing to have such similar code in two different locations.